multicluster/test/e2e/service_test.go

```diff
@@ -509,7 +515,8 @@ func (data *MCTestData) getNodeNamesFromCluster(clusterName string) (string, str
 		return "", "", fmt.Errorf("error when getting Node list: %v, stderr: %s", err, stderr)
 	}
 	nodes := strings.Split(strings.TrimRight(output, " "), " ")
-	gwIdx := rand.Intn(len(nodes)) // #nosec G404: for test only
+	mainRand := rand.New(rand.NewPCG(1, 0)) // Initialize with fixed seeds for deterministic behavior
+	gwIdx := mainRand.IntN(len(nodes))      // #nosec G404: for test only
```

Comment:
I don't know if deterministic behavior is really important / desired here. Maybe try with the main random generator provided by the package.

Overall I think maybe avoiding deterministic behavior is a good thing for e2e tests.